sql/core/src/test/scala/org/apache/spark/sql/test/SharedSQLContext.scala

```diff
@@ -61,8 +61,10 @@ trait SharedSQLContext extends SQLTestUtils with BeforeAndAfterEach with Eventua
    */
   protected override def beforeAll(): Unit = {
     SparkSession.sqlListener.set(null)
-    if (_spark == null) {
-      _spark = createSparkSession
+    synchronized {
```

Comment:
`beforeAll` ought to be called by one thread in the test framework before any tests start. I doubt this is the reason, but, maybe I don't have the full context here.